pkg/store/bucket.go

```diff
@@ -1122,20 +1126,23 @@ func (s *bucketBlockSet) getFor(mint, maxt, maxResolutionMillis int64) (bs []*bu
 	}
 
 	// Fill the given interval with the blocks for the current resolution.
-	// Our current resolution might not cover all data, so recursively fill the gaps with higher resolution blocks if there is any.
+	// Our current resolution might not cover all data, so recursively fill the gaps with higher resolution blocks
+	//if there is any.
 	start := mint
 	for _, b := range s.blocks[i] {
 		if b.meta.MaxTime <= mint {
 			continue
 		}
-		if b.meta.MinTime >= maxt {
+		// NOTE: Block intervals are half-open: [b.MinTime, b.MaxTime).
+		if b.meta.MinTime > maxt {
```

Comment:
@GiedriusS @brancz @squat 

I think this was tiny bug we had. Request for `Series` with time selection e.g `[mint=t1, mint=t1]` was matching no block. Now if there is any that starts with t1 it will be matched to return potentially single sample.

Reply:
Do you mean `[mint=t1,maxt=t1]`?

Reply:
Sorry, yes: Do you think that's fair?